openedx/adg/admin_dashboard/admin_settings/views.py

```diff
@@ -276,3 +276,39 @@ def get_queryset(self):
             )
 
         return invitations
+
+
+class UpdatePermissions(ListView):
+    """
+    UpdatePermissions View
+    """
+    def get(self, request, *args, **kwargs):
+        """
+        It will get invitation object of the user and give permissions which
+        are assigned to that user in invitation object
+
+        Returns:
+            It will redirect to admin dashboard if user is active and have permissions
+            otherwise it will redirect to home page
+        """
+        user = self.request.user
+        invitation_obj = Invitation.objects.filter(email=user.email).first()
```

Comment:
Let's cache the `business_unit__group` using `select_related` for the case of BU Admin invitation. It will save us two queries at line 301